Add a node.js min version check

People see for example `Error: Cannot find module 'fs/promises'` essentially random error messages when they upgrade and have an ancient Node.js.

We should add a minimum Node.js version check in https://github.com/SignalK/signalk-server/blob/master/bin/signalk-server with a clear prompt to upgrade as an error message.

--- a/bin/signalk-server
+++ b/bin/signalk-server
@@ -18,13 +18,15 @@
 
 const { engines } = require('../package.json');
 const semver = require('semver');
+const minimumVersion = engines.node;
+const recommendedVersion = "18";
 
-const version = engines.node;
-
-if (!semver.satisfies(process.version, version)) {
-  console.log(`The installed version of node (${process.version}) is older than the required minimum version (${version}). See https://github.com/SignalK/signalk-server/wiki/Updating-to-Node.js-18 for more information how to upgrade.`);
+if (!semver.satisfies(process.version, minimumVersion)) {
+  console.error(`The installed version of node (${process.version}) is older than the minimum required version (${minimumVersion}). See https://github.com/SignalK/signalk-server/wiki/Installing-and-Updating-Node.js for more information how to upgrade.`);
   process.exit(1);
-}
+} else if (!semver.satisfies(process.version, recommendedVersion)) {
+  console.warn(`The installed version of node (${process.version}) is different than the recommended version (${recommendedVersion}). See https://github.com/SignalK/signalk-server/wiki/Installing-and-Updating-Node.js for more information how to upgrade.`);
+} 
 
 const Server = require('../lib')
 const server = new Server()

--- a/bin/signalk-server
+++ b/bin/signalk-server
@@ -16,6 +16,16 @@
  * limitations under the License.
 */
 
+const { engines } = require('../package.json');
+const semver = require('semver');
+
+const version = engines.node;
+
+if (!semver.satisfies(process.version, version)) {
+  console.log(`The installed version of node (${process.version}) is older than the required minimum version (${version}). See https://github.com/SignalK/signalk-server/wiki/Updating-to-Node.js-18 for more information how to upgrade.`);
+  process.exit(1);
+}
+
 const Server = require('../lib')
 const server = new Server()
 

--- a/package.json
+++ b/package.json
@@ -60,7 +60,7 @@
   ],
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=18"
+    "node": ">=16"
   },
   "workspaces": [
     "packages/server-admin-ui-dependencies",

--- a/package.json
+++ b/package.json
@@ -60,7 +60,7 @@
   ],
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=16"
+    "node": ">=18"
   },
   "workspaces": [
     "packages/server-admin-ui-dependencies",
@@ -118,7 +118,7 @@
     "ncp": "^2.0.0",
     "node-fetch": "^2.6.0",
     "primus": "^7.0.0",
-    "semver": "^7.1.1",
+    "semver": "^7.5.4",
     "split": "^1.0.0",
     "stat-mode": "^1.0.0",
     "swagger-ui-express": "^4.5.0",

